modules/shell/command_test.go

```diff
@@ -19,3 +23,72 @@ func TestRunCommandAndGetOutput(t *testing.T) {
 	out := RunCommandAndGetOutput(t, cmd)
 	assert.Equal(t, text, strings.TrimSpace(out))
 }
+
+func TestRunCommandAndGetOutputOrder(t *testing.T) {
+	t.Parallel()
+
+	stderrText := "Hello, Error"
+	stdoutText := "Hello, World"
+	expectedText := "Hello, Error\nHello, World\nHello, Error\nHello, World\nHello, Error\nHello, Error"
+	bashCode := fmt.Sprintf(`
+echo_stderr(){
+	(>&2 echo "%s")
+}
+echo_stdout(){
+	echo "%s"
+}
+echo_stderr
+echo_stdout
+echo_stderr
+echo_stdout
+echo_stderr
+echo_stderr
+`,
+		stderrText,
+		stdoutText,
+	)
+	cmd := Command{
+		Command: "bash",
+		Args:    []string{"-c", bashCode},
+	}
+
+	out := RunCommandAndGetOutput(t, cmd)
+	assert.Equal(t, strings.TrimSpace(out), expectedText)
+}
+
+func TestRunCommandAndGetOutputConcurrency(t *testing.T) {
+	t.Parallel()
+
+	uniqueStderr := random.UniqueId()
+	uniqueStdout := random.UniqueId()
+
+	bashCode := fmt.Sprintf(`
+echo_stderr(){
+	sleep .$[ ( $RANDOM %% 10 ) + 1 ]s
+	(>&2 echo "%s")
+}
+echo_stdout(){
+	sleep .$[ ( $RANDOM %% 10 ) + 1 ]s
+	echo "%s"
+}
+for i in {1..5}
```

Comment:
Increase from 5 to something way bigger (500?). Make `sleep` short enough so the test doesn't take forever.

Reply:
Done

Reply:
Testing time was 2 seconds on my mac.

Reply:
Haha, OK, that's probably fast enough 😁